File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/FormContexts/Northwind.OrderDetailForm.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    public partial class OrderDetailForm : PrefixedContext
    {
        public OrderDetailForm(string idPrefix) : base(idPrefix) {}
    
        public IntegerEditor ProductID { get { return ById<IntegerEditor>("ProductID"); } }
        public DecimalEditor UnitPrice { get { return ById<DecimalEditor>("UnitPrice"); } }
        public StringEditor Quantity { get { return ById<StringEditor>("Quantity"); } }
        public StringEditor Discount { get { return ById<StringEditor>("Discount"); } }
    }
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.OrderDetailForm.ts
import { DecimalEditor, initFormType, IntegerEditor, PrefixedContext, ServiceLookupEditor } from "@serenity-is/corelib";

export interface OrderDetailForm {
    OrderID: IntegerEditor;
    ProductID: ServiceLookupEditor;
    UnitPrice: DecimalEditor;
    Quantity: IntegerEditor;
    Discount: DecimalEditor;
}

export class OrderDetailForm extends PrefixedContext {
    static readonly formKey = 'Northwind.OrderDetail';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!OrderDetailForm.init) {
            OrderDetailForm.init = true;

            initFormType(OrderDetailForm, [
                'OrderID', IntegerEditor,
                'ProductID', ServiceLookupEditor,
                'UnitPrice', DecimalEditor,
                'Quantity', IntegerEditor,
                'Discount', DecimalEditor
            ]);
        }
    }
}